style/javascript.md

```diff
@@ -536,6 +537,404 @@ to replace all uses of `var` in existing files.
+Becomes:
+```js
+{
+    ...dataToEventLogOnly,
+    ...dataToGraphite,
+    _request_id: KA.requestLogId,
+    _graphite_key_prefix: KA.gaeStatsKeyPrefix,
+    _graphite_keys: Object.keys(dataToGraphite).join()
+}
+```
+
+or:
+```js
+_.extend(defaultOptions, options || {})
+```
+
+Becomes:
+```js
+{...defaultOptions, ...options}
+```
+
+or:
+```js
+_.extend(json, this.model.toJSON())
+```
+
+Becomes:
+```js
+Object.assign(json, this.model.toJSON())
+```
+
+#### _.filter
+
+```js
+_.filter(resourceTimings, function(t) {
+    return nameRegexp.test(t.name);
+})
+```
+
+Becomes:
+```js
+resourceTimings.filter(t => nameRegexp.test(t.name))
+```
+
+#### _.has
+
+```js
+_.has(array, value)
+```
+
+Becomes:
+```js
+array.includes(value)
+```
+
+or:
+```js
+_.has(obj, value)
+```
+
+Becomes:
+```js
+value in obj
+```
+
+#### _.isArray
+
+```js
+_.isArray(someObj)
+```
+
+Becomes:
+```js
+Array.isArray(someObj)
+```
+
+#### _.isFunction
+
+```js
+_.isFunction(fn)
+```
+
+Becomes:
+```js
+typeof fn === "function"
+```
+
+#### _.isString
+
+```js
+_.isString(obj)
+```
+
+Becomes:
+```js
+typeof obj === "string"
+```
+
+#### _.keys
+
+```js
+_.keys(obj)
+```
+
+Becomes:
+```js
+Object.keys(obj)
+```
+
+#### _.last
+
+```js
+_.last(someArray)
+```
+
+Becomes:
+```js
+someArray[someArray.length - 1]
+```
+
+or (if you don't case about the array contents):
+```js
+someArray.pop()
+```
+
+#### _.map
+
+```js
+_.map(resourceTimings, function(t) {
+    return nameRegexp.test(t.name);
+})
+```
+
+Becomes:
+```js
+resourceTimings.map(t => nameRegexp.test(t.name))
+```
+
+#### _.max
+
+```js
+_.max(array)
+```
+
+Becomes:
+```js
+Math.max(...array)
+```
+
+#### _.omit
+
+```js
+_.omit(array, props)
+```
+
+Becomes:
+```js
+array.filter(prop => !props.includes(prop))
+```
+
+```js
+_.omit(obj, props)
+```
+
+Becomes:
+```js
+Object.keys(obj).reduce((result, prop) => {
+    if (!props.includes(prop)) {
+        result[prop] = attrs[prop];
+    }
+}, {})
+```
+
+#### _.once
+
+```js
+$(...).on("click", _.once(...))
+```
+
+Becomes:
+```js
+$(...).one("click", ...)
+```
+
+or:
+```js
+{
+    method: _.once(function() {
+        ...
+    })
+}
+```
+
+Becomes:
+```js
+{
+    method: function() {
+        // Only execute this, at most, once
+        if (this._initDone) {
+            return;
+        }
+
+        this._initDone = true;
+
+        ...
+    }
+}
+```
+
+or:
+```js
+var fetchMissionPercentages = _.once(function() {
+    return $.when(...).then(...);
+});
+```
+
+Becomes:
+```js
+var fetchMissionPercentages = function() {
+    let val = $.when(...).then(...);
+
+    // Override the function to avoid multiple calls
+    fetchMissionPercentages = () => val;
+
+    return val;
+};
+```
+
+#### _.object
+
+```js
+_.object(_.map(accentedChars, function(chars, c) {
+    return [c, "[" + c + c.toLowerCase() + chars + "]"];
+}))
+```
+
+Becomes:
+```js
+Object.entries(accentedChars).reduce((result, [c, chars]) => {
+    result[c] = `[${c}${c.toLowerCase()}${chars}]`;
```

Comment:
I think it's probably beneficial to limit each change to just one thing, so I wouldn't both move to reduce() and move to backticks in the same example.